File: web/src/services/reportsService.js
import api from './api'

export const reportsService = {
  // Get comprehensive business reports
  getBusinessReports: async (startDate, endDate) => {
    const params = {};
    if (startDate) params.startDate = startDate;
    if (endDate) params.endDate = endDate;
    
    const response = await api.get('/reports', { params })
    return response.data
  }
}

File: mobile/lib/services/reports_service.dart
import 'api_service.dart';

class ReportsService {
  final _api = ApiService();

  Future<Map<String, dynamic>?> getBusinessReports(String? startDate, String? endDate) async {
    try {
      String endpoint = '/reports';
      final params = <String, String>{};
      if (startDate != null) params['startDate'] = startDate;
      if (endDate != null) params['endDate'] = endDate;
      
      if (params.isNotEmpty) {
        final queryParams = params.entries.map((e) => '${e.key}=${e.value}').join('&');
        endpoint += '?$queryParams';
      }
      
      final response = await _api.get(endpoint);
      if (response['success'] == true) {
        return response['data'];
      }
      return null;
    } catch (e) {
      return null;
    }
  }
}

